### DISCUSSION TITLE
await-async-query rule with Cypress

### DISCUSSION THREAD

**Discussion by MerlinMason** [testing-library/eslint-plugin-testing-library#533] [Q&A] [answered]
Is the rule `await-async-query` still valid when using testing library with Cypress?

It seems to contradict [Cypress's guidance](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-async-tests.md) on not using `async` in tests.

**Comment by MichaelDeBoey** [comment, marked as answer]
This plugins isn't officially supporting any other testing library than `dom`, `angular`, `react` or `vue`, so it could happen that there are some inconsistencies.

We already have #80 to track supporting more testing libraries

**Reply by MerlinMason** [reply]
Ahh ok, thanks for clarifying! 

I'm using Jest and Cypress in my project, so I've added the following to my eslint config:

```js
overrides: [
    {
        files: ["**/cypress/**/*.spec.js", "**/cypress/**/*.spec.ts"],
        rules: {
            "testing-library/await-async-query": 0,
        },
    },
],
```

**Reply by Belco90** [reply]
Hey @MerlinMason! Since `eslint-plugin-testing-library` isn't fully compatible with Cypress, and `cypress-testing-library` offers a restricted group of features from Testing Library (IIRC just `within` and `findBy*` queries), I would suggest enabling `eslint-plugin-testing-library` only for your non-cypress tests. You can find an example of how to do this for jest here: https://github.com/testing-library/eslint-plugin-testing-library#eslint-overrides
